modules/cpmso-openstack-without-az-config.adoc

```diff
@@ -0,0 +1,74 @@
+// Module included in the following assemblies:
+//
+// * machine_management/control_plane_machine_management/cpmso-troubleshooting.adoc
+
+:_content-type: PROCEDURE
+[id="cpmso-openstack-ts-az-config_{context}"]
+= Configuring {rh-openstack} clusters without availability zones
+
+If you upgrade a cluster on {rh-openstack-first} from version 4.13 to 4.14 that uses root volumes with availability zones, you must manually configure machine plane machine resources to use control plane machine sets.
+
+If you are starting with the {product-title} cluster, version 4.14, this process is done automatically. If you are  upgrading, follow the steps.
+
+This makes the root volume availability zone to be taken from the compute availability zone, which is not available in cinder. The control plane machine cannot be created. 
+
+For the clusters deployed before the {product-title} version 4.14, you need to manually update the control plane machine resources, so that they are updated on the instances.
+
+[NOTE]
+====
+Manually editing control plane machine resources does not update the control plane instances. However, it is necessary in order for the `cluster-control-plane-machine-set-operator` to correctly generate a `ControlPlaneMachineSet` for your cluster.
+====
+
+. Update the `ProviderSpec`, for all the control plane machines matching the environment, and set the property `rootVolume.availabilityZone` of `spec.providerSpec` to the value of the actual volume AZ (example with `master-0`):
++
+[source,terminal]
+----
+$ oc edit machine/<cluster_id>-master-0 -n openshift-machine-api
+<make edits>
+----
++
+[source,yaml]
+----
+providerSpec:
+  value:
+    apiVersion: machine.openshift.io/v1alpha1
+    availabilityZone: az0
+      cloudName: openstack
+    cloudsSecret:
+      name: openstack-cloud-credentials
+      namespace: openshift-machine-api
+    flavor: m1.xlarge
+    image: rhcos-4.14
+    kind: OpenstackProviderSpec
+    metadata:
+      creationTimestamp: null
+    networks:    
+    - filter: {}
+      subnets:  
+      - filter:
+          name: refarch-lv7q9-nodes
+          tags: openshiftClusterID=refarch-lv7q9
+    rootVolume:
+        availabilityZone: nova <1>
+        diskSize: 30
+        sourceUUID: rhcos-4.12
+        volumeType: fast-0
+    securityGroups:
+    - filter: {}
+      name: refarch-lv7q9-master
+    serverGroupName: refarch-lv7q9-master
+    serverMetadata:
+      Name: refarch-lv7q9-master
+      openshiftClusterID: refarch-lv7q9
+    tags:
+    - openshiftClusterID=refarch-lv7q9
+    trunk: true
+    userDataSecret:
+      name: master-user-data
+----
+<1> Add the zone name.
+
+. In case you updated your control plane machine resources after install, follow these steps. 
+* In your {rh-openstack} cluster, find the AZ of the root volumes for your control plane machines and set it in the `availabilityZone` property of all the control plane machines.
```

Comment:
I believe this is meant to be mutually exclusive from the first step in the solution--if you configured yourself out of following the happy path, look up what you need in OpenStack and put it in the `providerSpec`.

Reply:
Let me change that.